From 56b970fc27f68a3a58c660941f40595aa73fa821 Mon Sep 17 00:00:00 2001
From: Sam Nelson <sam.nelson@ti.com>
Date: Tue, 20 Oct 2015 07:22:28 -0400
Subject: [PATCH 12/12] dts: k2e: Add entries for DSP remote proc nodes for k2e
 platform

- Add entries for common memory pool to be used for Remote proc
- Add reserved area for dsp code and data usage
- Add entries for DSP remot proc nodes & common memory node

Signed-off-by: Sam Nelson <sam.nelson@ti.com>
---
 arch/arm/boot/dts/k2e-evm.dts | 31 +++++++++++++++++++++++++++++++
 arch/arm/boot/dts/k2e.dtsi    | 27 +++++++++++++++++++++++++++
 2 files changed, 58 insertions(+)

diff --git a/arch/arm/boot/dts/k2e-evm.dts b/arch/arm/boot/dts/k2e-evm.dts
index a8b12a7..0c529c9 100644
--- a/arch/arm/boot/dts/k2e-evm.dts
+++ b/arch/arm/boot/dts/k2e-evm.dts
@@ -16,6 +16,26 @@
 	compatible = "ti,k2e-evm", "ti,k2e", "ti,keystone";
 	model = "Texas Instruments Keystone 2 Edison EVM";
 
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		dsp_common_cma_pool: dsp_common_cma_pool {
+			compatible = "shared-dma-pool";
+			reg = <0x00000008 0x20000000 0x00000000 0x800000>;
+			reusable;
+			status = "okay";
+		};
+
+		dsp_reserved_code_area: dsp_reserved_code_area {
+			compatible = "shared-dma-pool";
+			reg = <0x00000008 0x20800000 0x00000000 0x1F7FFFFF>;
+			no-map;
+			status = "okay";
+		};
+	};
+
 	soc {
 
 		clocks {
@@ -160,3 +180,14 @@
 &pcie1 {
 	status = "okay";
 };
+
+&dsp0 {
+	status = "okay";
+	memory-region = <&dsp_common_cma_pool>;
+};
+
+&uio_dspmem {
+	status = "okay";
+	mem  = <0x0c000000 0x000600000
+		0xa0000000 0x20000000>;
+};
diff --git a/arch/arm/boot/dts/k2e.dtsi b/arch/arm/boot/dts/k2e.dtsi
index ac30527..149b9f3 100644
--- a/arch/arm/boot/dts/k2e.dtsi
+++ b/arch/arm/boot/dts/k2e.dtsi
@@ -43,6 +43,10 @@
 		};
 	};
 
+	aliases {
+		rproc0 = &dsp0;
+	};
+
 	soc {
 		/include/ "k2e-clocks.dtsi"
 
@@ -89,6 +93,29 @@
 			gpio,syscon-dev = <&devctrl 0x240>;
 		};
 
+		dsp0: dsp0 {
+			compatible = "ti,keystone-rproc";
+			reg = <0x02620844 4
+			       0x0235083c 4
+			       0x02350a3c 4
+			       0x02620240 4>;
+			reg-names = "boot-address", "psc-mdstat", "psc-mdctl",
+				"ipcgr";
+			mem = <0x10e00000 0x00008000
+			       0x10f00000 0x00008000
+			       0x10800000 0x00080000>;
+			interrupt-parent = <&kirq0>;
+			interrupts = <0 8>;
+			kick-gpio = <&dspgpio0 27 0>;
+			clocks = <&clkgem0>;
+			status = "disabled";
+		};
+
+		uio_dspmem: dspmem {
+			compatible = "ti,uio-ti-pdrv";
+			status = "disabled";
+		};
+
 		pcie1_phy: pciephy@2326000 {
 			#phy-cells = <0>;
 			compatible = "ti,keystone-serdes-phy";
-- 
1.9.1

